src/modules/account/account.service.ts

```diff
@@ -78,9 +81,19 @@ export class AccountService {
       throw new NotFoundException(businessException([ERRORS.account.notFound]))
     }
 
+    const accountOldName = accountFetched.name
     const accountMapped = AccountMapper.toUpdateEntity(accountFetched, data)
     const accountSaved = await this.accountRepository.save(accountMapped)
 
+    if (data.name) {
+      await this.instructionElementRepository
+        .createQueryBuilder()
+        .update(InstructionElementEntity)
+        .set({ genericType: data.name })
+        .where({ genericType: accountOldName })
```

Comment:
Is it possible to update by id instead of name?

Reply:
No, because in InstructionElementEntity the 'genericType' is stored as string.